gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -273,42 +276,57 @@ private boolean isDirectPathEnabled() {
     return false;
   }
 
+  private boolean isDirectPathXdsEnabledViaBuilderOption() {
+    return Boolean.TRUE.equals(attemptDirectPathXds);
+  }
+
+  private boolean isDirectPathXdsEnabledViaEnv() {
+    String directPathXdsEnv = envProvider.getenv(DIRECT_PATH_ENV_ENABLE_XDS);
+    return Boolean.parseBoolean(directPathXdsEnv);
+  }
+
   @InternalApi
   public boolean isDirectPathXdsEnabled() {
     // Method 1: Enable DirectPath xDS by option.
-    if (Boolean.TRUE.equals(attemptDirectPathXds)) {
+    if (isDirectPathXdsEnabledViaBuilderOption()) {
```

Comment:
You can just return isDirectPathXdsEnabledViaBuilderOption() || isDirectPathXdsEnabledViaEnv().

Reply:
Done.